src/main/java/seedu/address/model/person/UniqueTutorialTagList.java

```diff
@@ -0,0 +1,129 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Iterator;
+import java.util.List;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.person.exceptions.DuplicateTutorialTagException;
+import seedu.address.model.person.exceptions.TutorialTagNotFoundException;
+import seedu.address.model.tag.TutorialTag;
+
+
+/**
+ * A list of tutorialTags that enforces uniqueness between its elements and does not allow nulls.
+ * A tutorialTag is considered unique by comparing using {@code TutorialTag#isSameTutorialTag(TutorialTag)}.
+ * As such, adding and updating of tutorialTags uses TutorialTag#isSameTutorialTag(TutorialTag) for equality so as to
+ * ensure that the tutorialTag being added or updated is unique in terms of identity in the UniqueTutorialTagList.
+ * However, the removal of a tutorialTag uses TutorialTag#equals(Object) so as to ensure that the tutorialTag
+ * with exactly the same TagName will be removed.
+ *
+ * Supports a minimal set of list operations.
+ */
+public class UniqueTutorialTagList implements Iterable<TutorialTag> {
+    private final ObservableList<TutorialTag> internalList = FXCollections.observableArrayList();
+    private final ObservableList<TutorialTag> internalUnmodifiableList =
+            FXCollections.unmodifiableObservableList(internalList);
+
+    /**
+     * Returns true if the list contains an equivalent TutorialTag as the given argument.
+     */
+    public boolean contains(TutorialTag toCheck) {
+        requireNonNull(toCheck);
+        return internalList.stream().anyMatch(toCheck::isSameTutorialTag);
+    }
+
+    /**
+     * Adds a TutorialTag to the list.
+     * The TutorialTag must not already exist in the list.
+     */
+    public void add(TutorialTag toAdd) {
+        requireNonNull(toAdd);
+        if (contains(toAdd)) {
+            throw new DuplicateTutorialTagException();
+        }
+        internalList.add(toAdd);
+    }
+
+    /**
+     * Removes the equivalent TutorialTag from the list.
+     * The TutorialTag must exist in the list.
+     */
+    public void remove(TutorialTag toRemove) {
+        requireNonNull(toRemove);
+        if (!internalList.remove(toRemove)) {
+            throw new TutorialTagNotFoundException();
+        }
+    }
+
+    public void setTutorialTags(UniqueTutorialTagList replacement) {
+        requireNonNull(replacement);
+        internalList.setAll(replacement.internalList);
+    }
```

Comment:
I would to clarify the use of this method

Reply:
It is to remove TutorialTags from UniqueTutorialTagList

Reply:
Thank you for the clarification!